Src/Geneva.c

```diff
@@ -93,9 +87,12 @@ void geneva_Callback(){
 		}
 	if(geneva_idle != geneva_state){
 		float state = geneva_Encodervalue();
-		int PIDoutput = singlePID(Geneva_pid.ref, state, Geneva_pid.K_terms, Geneva_pid.timestep);
-		PIDoutput = ClipFloat(PIDoutput, -4 * Geneva_pid.max_pwm, 4 * Geneva_pid.max_pwm); //Limit the PID output, legacy, not sure if necessary
-		pid_SetOutput(PIDoutput, &Geneva_pid); //send signal to the motor
+		float err = Geneva_pid.ref- state;
+		if(abs(err)>GENEVA_MAX_ALLOWED_OFFSET*0.5){ //prevents the integrate control from oscillating around zero, and heating the driver
+			float PIDoutput = PID(err, Geneva_pid.PIDvar); //PID from control_util.h
+			PIDoutput = ClipFloat(PIDoutput, -4 * Geneva_pid.max_pwm, 4 * Geneva_pid.max_pwm); //Limit the PID output, legacy, not sure if necessary
```

Comment:
magic numbers? Was more based on the previous code, where we added P and I which were both clipped by 2times max_pwm.  so combined the range is 4*max_pwm?

Reply:
Just checked the old code, there are a lot of hacks that really make the control useless. It is better to base this of something done by yourself rather than the old code. Because a lot of random stuff can happen there

Reply:
well it matters not really it worked in both cases